reconcile/utils/terrascript_client.py

```diff
@@ -1536,17 +1536,72 @@ def populate_tf_resource_s3(self, resource, namespace_info):
+            if prefix is not None:
+                notification['queue'][0]['filter_prefix'] = \
+                    prefix
+            suffix = common_values.get('filter_suffix', None)
+            if suffix is not None:
+                notification['queue'][0]['filter_suffix'] = \
+                    suffix
 
             notification_tf_resource = aws_s3_bucket_notification(
-                sqs_identifier, **notification_values)
+                sqs_identifier, **notification)
+            tf_resources.append(notification_tf_resource)
+        event_notifications = common_values.get('event_notifications')
+
+        for event_notification in event_notifications:
+            destination_type = event_notification['destination_type']
+            detination_identifier = event_notification['destination']
+            event_type = event_notification['event_type']
+
+            if detination_identifier is not None:
+                identifier_is_arn = True \
+                    if detination_identifier.startswith('arn:') else False
+            if identifier_is_arn:
+                resource_name = detination_identifier.split(':')[-1]
+                resource_arn = detination_identifier
+            else:
+                resource_name = detination_identifier
+                resource_arn = '${data.aws_sns_topic.' \
+                    + detination_identifier + '.arn}'
+
+            resource_values = {
+                'name': resource_name
+            }
+            resource_provider = values.get('provider')
+            if resource_provider:
+                resource_values['provider'] = resource_provider
+
+            if destination_type == 'sns':
+                resource_data = data.aws_sns_topic(resource_name,
+                                                   **resource_values)
+                notification_type = 'topic'
+            elif destination_type == 'sqs':
+                resource_data = data.aws_sqs_queue(resource_name,
+                                                   **resource_values)
+                notification_type = 'queue'
+
+            tf_resources.append(resource_data)
+            s3_events = event_type
+            notification = {
+                'bucket': '${' + bucket_tf_resource.id + '}',
+                notification_type: [{
+                    'id': resource_name,
+                    notification_type+'_arn': resource_arn,
+                    'events': s3_events
+                }]
+            }
+
+            prefix = event_notification.get('filter_prefix', None)
+            if prefix is not None:
+                notification[notification_type][0]['filter_prefix'] = prefix
+            suffix = event_notification.get('filter_suffix', None)
+            if suffix is not None:
+                notification[notification_type][0]['filter_suffix'] = suffix
+
+            notification_tf_resource = aws_s3_bucket_notification(
+                identifier+'-'+resource_name, **notification)
```

Comment:
Ideally, we can append all notification into one aws_s3_bucket_notification resource with the values of
```
            notification = {
                'bucket': '${' + bucket_tf_resource.id + '}',
                'topic': [list of SNS Topic],
                'queue': [list of SQS Queue]
            }

```
I think this will be more clear, wdyt?

Reply:
Good point, I'll make this part explicit and a list that can be appended.